server/src/modules/users/users.service.ts

```diff
@@ -1,55 +1,74 @@
-import { HttpException, Injectable } from '@nestjs/common';
-import { User, UserAuth, Prisma, Profile, MapRank, Map as MapDB } from '@prisma/client';
-import { UserDto } from '../../@common/dto/user/user.dto';
-import { ProfileDto, UserProfileDto } from '../../@common/dto/user/profile.dto';
+import { HttpException, Injectable, NotFoundException } from '@nestjs/common';
+import { Follow, Map as MapDB, MapRank, Prisma, User, UserAuth } from '@prisma/client';
+import { UpdateUserDto, UserDto } from '../../@common/dto/user/user.dto';
+import { ProfileDto, ProfileUpdateDto } from '../../@common/dto/user/profile.dto';
 import { PagedResponseDto } from '../../@common/dto/common/api-response.dto';
 import { UsersRepo } from './users.repo';
-import { appConfig } from 'config/config';
+import { appConfig } from '../../../config/config';
 import { lastValueFrom, map } from 'rxjs';
 import * as xml2js from 'xml2js';
 import { HttpService } from '@nestjs/axios';
 import { ActivityDto } from '../../@common/dto/user/activity.dto';
 import { FollowerDto } from '../../@common/dto/user/followers.dto';
 import { MapRankDto } from '../../@common/dto/map/mapRank.dto';
-import { UserRunDto } from '../../@common/dto/run/runs.dto';
 import { UserMapCreditDto } from '../../@common/dto/map/mapCredit.dto';
+import { EBan } from '../../@common/enums/user.enum';
+import { EActivityTypes } from '../../@common/enums/activity.enum';
+import { RunDto } from '../../@common/dto/run/runs.dto';
 
 @Injectable()
 export class UsersService {
     constructor(private readonly userRepo: UsersRepo, private readonly http: HttpService) {}
 
     //#region GETs
-    public async GetAll(skip?: number, take?: number): Promise<PagedResponseDto<UserDto[]>> {
-        const dbResponse = await this.userRepo.GetAll(undefined, skip, take);
 
-        const totalCount = dbResponse[1];
-        const users = dbResponse[0];
+    // TODO: mapRank
+    public async GetAll(
+        skip?: number,
+        take?: number,
+        expand?: string[],
+        search?: string,
+        playerID?: string,
+        playerIDs?: string[]
+    ): Promise<PagedResponseDto<UserDto[]>> {
+        const where: Prisma.UserWhereInput = {};
+
+        if (playerID) {
```

Comment:
Potential bug here, what happens when you provide a playerID and playerIDs? Depending on the order by in the repo you may not always get the same response. I'd swap these around, too
```ts
if (playerIDs) {
    where.steamID = { in: playerIDs };
} else if (playerID) {
   take = 1;
    where.steamID = playerID;
}
```
As then if you somehow provide both the more specific request is always honoured (as its quicker). Or just throw a bad request error if a client provides both

Reply:
Dunno why I can't respond to the logger comment, but that logging stuff is on your code, it's already on the main branch. It's just random debug, I wanted to use it and Logger.log doesn't seem to work with Jest currently. I assume it'll get deleted in the future by you, I wouldn't worry about it

Reply:
Yeah, should just throw a bad request. Made it do that in last push